src/hgraph/_wiring/_wiring_node_instance.py

```diff
@@ -80,7 +83,14 @@ class WiringNodeInstance:
     error_handler_registered: bool = False
     trace_back_depth: int = 1  # TODO: decide how to pick this up, probably via the error context?
     capture_values: bool = False
-    _hash: int | None = None
+    _treat_as_source_node: bool = False
+    rank_marker: frozendict[str, "WiringNodeinstance"] = field(default_factory=frozendict)
```

Comment:
This is the new rank_marker that allows to specify phantom dependencies used for ranking operations only.